src/Client/DiscoveryPolicy.cs

```diff
@@ -19,7 +22,25 @@ public class DiscoveryPolicy
         /// <summary>
         /// Method of comparison for issuer and authority names. Defaults to <see cref="StringComparison.Ordinal" />
         /// </summary>
-        public StringComparison AuthorityNameComparison { get; set; } = StringComparison.Ordinal;
+        [Obsolete("Please use AuthorityValidationStrategy")]
+        public StringComparison AuthorityNameComparison
+        {
+            get
+            {
+                return _authorityNameComparison;
+            }
+            set
+            {
+                _authorityNameComparison = value;
+                AuthorityValidationStrategy = new StringComparisonAuthorityValidationStrategy(value);
+            }
+        }
+
+        /// <summary>
+        /// Strategy used to validate issuer name and endpoints based on expected authority.
+        /// Defaults to <see cref="StringComparisonAuthorityValidationStrategy.Default"/>.
+        /// </summary>
+        public IAuthorityValidationStrategy AuthorityValidationStrategy { get; set; } = new StringComparisonAuthorityValidationStrategy(defaultAuthorityNameComparison);
```

Comment:
Here you are newing up the strategy - instead of using Instance. I'd prefer consistency.

Reply:
I used "Instance" with the non parameterizable class AuthorityUrlValidationStrategy.
The class StringComparisonAuthorityValidationStrategy is parameterizable with StringComparison. IMHO it's consistent, but it maybe is a matter of taste.